Go parts of `quicktest.sh` and `lit` tests fail because of lack of `GO111MODULE=off`

macOS 11.2.3
Dafny 3.0.0-30204 (built from source at commit d25ae2577b5d38b3b4df90a7e8062d3f6fbfd2a7)
golang 1.16.2

quicktests.sh ended in failure like this:
```
...
Compiling to Go

Dafny program verifier finished with 0 verified, 0 errors
c-go/src/c.go:6:3: package System_ is not in GOROOT (/usr/local/Cellar/go/1.16.2/libexec/src/System_)
c-go/src/c.go:5:3: package dafny is not in GOROOT (/usr/local/Cellar/go/1.16.2/libexec/src/dafny)
```
(which, by the way, as a go hobbyist and dafny noob, was hard for me to ascertain as failure or warning or what)

I could fix it with `export GO111MODULE=off`.

A number of `lit` tests were also failing, and the `export` wasn't helping. I could fix it with
```
go env -w GO111MODULE=off
```
... though IIUC this might interfere with other Go projects. I don't have any in flight so I don't mind.

Relatedly, in my 6-Core Core i7 Macbook, the lit tests run ~3x faster if I limit the workers with `lit -j 8 .`.

--- a/Source/Dafny/Compiler-go.cs
+++ b/Source/Dafny/Compiler-go.cs
@@ -3443,6 +3443,9 @@ namespace Microsoft.Dafny {
         RedirectStandardError = false,
       };
       psi.EnvironmentVariables["GOPATH"] = GoPath(targetFilename);
+      // Dafny compiles to the old Go package system, whereas Go has moved on to a module
+      // system. Until Dafny's Go compiler catches up, the GO111MODULE variable has to be set.
+      psi.EnvironmentVariables["GO111MODULE"] = "auto";
 
       try {
         using (var process = Process.Start(psi)) {

--- a/Test/comp/manualcompile/lit.local.cfg
+++ b/Test/comp/manualcompile/lit.local.cfg
@@ -26,5 +26,6 @@ config.substitutions.append( ('%binaryDir', binaryDir) )
 current_dir = os.path.join(config.test_source_root, 'comp', 'manualcompile')
 
 config.environment['GOPATH'] = os.path.join(current_dir, 'ManualCompile-go')
+config.environment['GO111MODULE'] = 'auto'
 
 config.environment['CLASSPATH'] = os.path.join(current_dir, 'ManualCompile-java') + ":" + os.path.join( binaryDir, "DafnyRuntime.jar" )

